src/posix/platform/resolver.cpp

```diff
@@ -95,16 +97,25 @@ void Resolver::LoadDnsServerListFromConf(void)
 
     while (fp.good() && std::getline(fp, line) && mUpstreamDnsServerCount < kMaxUpstreamServerCount)
     {
-        if (line.find(kNameserverItem, 0) == 0)
+        if (line.find(kNameserverItem, 0))
```

Comment:
You should check it against `npos` instead of zero.

See https://en.cppreference.com/w/cpp/string/basic_string/find